### PR TITLE
Add comments and update callback arguments' names

### PR DESCRIPTION
**Pull request by gerhynes**
I've added a few comments to the solution and given the callback arguments more descriptive names to clarify what the solution is doing, since this challenge seemed a little more complex than the previous challenges.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #XXXXX

### CODE REVIEW

File: guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data/index.md [hunk 1]
```diff
@@ -120,11 +120,19 @@ var watchList = [
...
-
-var averageRating = watchList.filter(x => x.Director === "Christopher Nolan").map(x => Number(x.imdbRating)).reduce((x1, x2) => x1 + x2) / watchList.filter(x => x.Director === "Christopher Nolan").length;
+var averageRating = 
+    watchList
+    // Use filter to find films directed by Christopher Nolan
+      .filter(film => film.Director === "Christopher Nolan")
+      // Use map to convert their ratings from strings to numbers
+      .map(film => Number(film.imdbRating))
+      // Use reduce to add together their ratings
+      .reduce((rating1, rating2) => rating1 + rating2)
```
Comment (RandellDawson) [inline guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data/index.md]: While we are making parameter names more readable, the first parameter represents the accumulator so:
```suggestion
      .reduce((sumOfRatings, rating) => sumOfRatings + rating) 
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@GK-Hynes Your changes are good, but I have one recommendation.

**Comment by RandellDawson** [comment]
After reviewing this again, I realize it is not using two-space indentation.  Also, I think find it better if the watchList variable is on the same line of the `=`.
```js
var averageRating = watchList
  // Use filter to find films directed by Christopher Nolan
  .filter(film => film.Director === "Christopher Nolan")
  // Use map to convert their ratings from strings to numbers
  .map(film => Number(film.imdbRating))
  // Use reduce to add together their ratings
  .reduce((sumOfRatings, rating) => sumOfRatings + rating) 
  // Divide by the number of Nolan films to get the average rating 
  / watchList.filter(film => film.Director === "Christopher Nolan").length;
```

**Comment by GK-Hynes** [comment]
Sorry about the indentation discrepancy. Having watchlist on the same line as the `=` also seems sensible.

**Review by RandellDawson — APPROVED** [review]
LGTM

**Comment by GK-Hynes** [comment]
Thanks for helping me to make it a better contribution.

**Pull request merged**